airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -652,3 +654,30 @@ def delete_objects(self, bucket, keys):
             if "Errors" in response:
                 errors_keys = [x['Key'] for x in response.get("Errors", [])]
                 raise AirflowException("Errors when deleting: {}".format(errors_keys))
+
+    @provide_bucket_name
+    @unify_bucket_name_and_key
+    def download_file(self, key: str, bucket_name: Optional[str] = None, local_path: str = '/tmp') -> str:
```

Comment:
Instead of hardcoding the `/tmp` dir. We could ask this using a Python function, which is system independent since it will ask the OS for a temp dir: https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

Reply:
Actually if I pass `dir=None` to the `NamedTemporaryFile` constructor it will use `tempfile.gettempdir()`. 👍 

In `NamedTemporaryFile`:
`prefix, suffix, dir, output_type = _sanitize_params(prefix, suffix, dir)`

In `_sanitize_params`:
```
    if dir is None:
        if output_type is str:
            dir = gettempdir()
        else:
            dir = gettempdirb()
```